File: backend/src/main/java/com/coigniez/resumebuilder/domain/column/ColumnResponse.java
package com.coigniez.resumebuilder.domain.column;

import java.util.List;

import com.coigniez.resumebuilder.domain.columnsection.ColumnSectionResponse;
import com.coigniez.resumebuilder.domain.layout.enums.ColorLocation;

import jakarta.validation.constraints.NotNull;
import lombok.*;


@Data
@Builder
public class ColumnResponse {

    @NotNull
    private long id;
    @NotNull
    private int columnNumber;
    @NotNull
    private List<ColumnSectionResponse> sectionMappings;

    @NotNull
    private ColorLocation backgroundColor;
    @NotNull
    private ColorLocation textColor;
    @NotNull
    private ColorLocation borderColor;

    @NotNull
    private double paddingLeft;
    @NotNull
    private double paddingRight;
    @NotNull
    private double paddingTop;
    @NotNull
    private double paddingBottom;

    @NotNull
    private double borderLeft;
    @NotNull
    private double borderRight;
    @NotNull
    private double borderTop;
    @NotNull
    private double borderBottom;
}

File: frontend/src/app/services/models/column-response.ts
/* tslint:disable */
/* eslint-disable */
import { ColumnSectionResponse } from '../models/column-section-response';
export interface ColumnResponse {
  backgroundColor?: 'PRIMARY' | 'SECONDARY' | 'ACCENT' | 'DARK_BG' | 'LIGHT_BG' | 'DARK_TEXT' | 'LIGHT_TEXT';
  borderBottom?: number;
  borderColor?: 'PRIMARY' | 'SECONDARY' | 'ACCENT' | 'DARK_BG' | 'LIGHT_BG' | 'DARK_TEXT' | 'LIGHT_TEXT';
  borderLeft?: number;
  borderRight?: number;
  borderTop?: number;
  columnNumber?: number;
  id?: number;
  paddingBottom?: number;
  paddingLeft?: number;
  paddingRight?: number;
  paddingTop?: number;
  sectionMappings?: Array<ColumnSectionResponse>;
  textColor?: 'PRIMARY' | 'SECONDARY' | 'ACCENT' | 'DARK_BG' | 'LIGHT_BG' | 'DARK_TEXT' | 'LIGHT_TEXT';
}
